src/core/ext/filters/client_channel/retry_filter.cc

```diff
@@ -1693,6 +1693,8 @@ void RetryFilter::CallData::CallAttempt::BatchData::RunClosuresForCompletedCall(
 
 void RetryFilter::CallData::CallAttempt::BatchData::RecvTrailingMetadataReady(
     void* arg, grpc_error_handle error) {
+  RefCountedPtr<grpc_call_stack> owning_call_stack =
```

Comment:
I can try... ASAN didn't report any faults FWIW, but TSAN was unhappy.
https://source.cloud.google.com/results/invocations/5b122974-4977-4862-beb1-dc1af9fbbd1d/targets/%2F%2Ftest%2Fcore%2Fend2end:h2_census_test@max_concurrent_streams@poller%3Dpoll/log contains an example failure.

I spent a few evenings tracking this down, and this fixes the problem in code that I'm hoping to get deleted in short order, so unless there's strong hesitations about safety I'd suggest let's proceed.

Reply:
Huh, that's really weird.  I don't understand that TSAN failure at all.  It looks like it's saying that the code in the retry filter was running before the call is unreffed, but then subsequently unreffing the call should be exactly the type of synchronization change that should prevent TSAN from complaining.

Not understanding this makes me fairly uncomfortable, but I guess I agree that it's not worth putting more time into it, since we're going to rewrite this in the not-too-distant future.  For now, please add a comment explaining that this ref is being added to avoid a TSAN failure that we don't really understand but aren't going to put time into diagnosing right now.